chronos/asyncmacro2.nim

```diff
@@ -81,6 +81,27 @@ proc asyncSingleProc(prc: NimNode): NimNode {.compileTime.} =
 
   let prcName = prc.name.getName
 
+  var
+    possibleExceptions = nnkBracket.newTree(newIdentNode("CancelledError"))
+    possibleExceptionsTuple = nnkTupleConstr.newTree(newIdentNode("CancelledError"))
+    foundRaises = -1
+
+  when (NimMajor, NimMinor) < (1, 4):
+    possibleExceptions.add(newIdentNode("Defect"))
+
+  for index, pragma in pragma(prc):
+    if pragma.kind == nnkExprColonExpr and pragma[0] == ident "raises":
+      foundRaises = index
+      for possibleRaise in pragma[1]:
```

Comment:
shouldn't this be `pragma[foundRaises]`?

Reply:
nvm, this is correct, you're iterating over all the pragmas...

Reply:
The ast returned by pragma looks like that:
```
Pragma
  Ident "gcsafe"
  ExprColonExpr
    Ident "raises"
    Bracket
      Ident "ValueError"
      Ident "CatchableError"
```

The first loop is responsible to find the "raises" pragma, and store it's index in the `foundRaises` (to be deleted later)
Next, we iterate over each exception in the Bracket node, which is `pragma[1]`